src/Components/Facility/Consultations/DailyRoundsFilter.tsx

```diff
@@ -0,0 +1,129 @@
+import { Popover, Transition } from "@headlessui/react";
+import ButtonV2 from "../../Common/components/ButtonV2";
+import { Fragment } from "react";
+import { SelectFormField } from "../../Form/FormFields/SelectFormField";
+import TextFormField from "../../Form/FormFields/TextFormField";
+import CareIcon from "../../../CAREUI/icons/CareIcon";
+import dayjs from "dayjs";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { DailyRoundsFilterModel } from "../models";
+
+export const DailyRoundsFilter = (props: {
+  setFilter: (data: DailyRoundsFilterModel) => void;
+}) => {
+  const [filter, setFilter] = useState({
+    roundType: "",
+    fromDateTime: "",
+    toDateTime: "",
+  });
+
+  const clearFilter = () => {
+    const clearedFilter = {
+      roundType: "",
+      fromDateTime: "",
+      toDateTime: "",
+    };
```

Comment:
Use `undefined` / `null` to set no filter applied! `useQuery`/`request` will automatically remove such query params from being serialized in the first place.

```suggestion
    const clearedFilter = {
      roundType: undefined,
      fromDateTime: undefined,
      toDateTime: undefined,
    };
```